packages/modal/src/Modal.js

```diff
@@ -0,0 +1,186 @@
+import React from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import { color, width, height } from 'styled-system'
+import { Transition, config } from 'react-spring'
+import { DialogOverlay, DialogContent } from '@reach/dialog'
+import { Box, CloseButton, Flex } from 'pcln-design-system'
+
+const Overlay = styled(DialogOverlay)`
+  background-color: rgba(0, 0, 0, 0.7);
+  bottom: 0;
+  left: 0;
+  overflow-y: auto;
+  position: fixed;
+  right: 0;
+  top: 0;
+  z-index: ${props => props.zindex || 100};
+  font-family: ${props => props.theme.font};
+`
+
+const Dialog = styled(DialogContent)`
+  ${color} 
+  ${width}
+  ${height}
+  max-width: calc(100vw - 32px);
+  position: relative;
+  margin-left: auto;
+  margin-right: auto;
+  box-shadow: ${props => props.theme.boxShadows[3]};
+  &:focus {
+    outline: none;
+  }
+  max-height: 100vh;
+  ${props =>
+    props.enableoverflow &&
+    `
+    max-height: none;
+  `}
+`
+
+const HeaderWrapper = styled.div`
+  flex-shrink: 0;
+`
+
+const FloatCloseButton = styled(CloseButton)`
+  position: absolute;
+  top: 16px;
+  right: 16px;
+  z-index: 2;
+
+  svg {
+    vertical-align: top;
+  }
+
+  &:focus {
+    outline: none;
+  }
+`
+
+const ContentWrapper = styled(Box)`
+  position: relative;
+  ${props => {
+    if (!props.enableoverflow) {
+      return `
+        overflow-y: scroll;
+        overflow-x: hidden;
+        height: 100%;
+        -webkit-overflow-scrolling: touch;
+      `
+    }
+  }}
+`
+
+const OverlayWrapper = styled.div`
+  display: table;
+  height: 100%;
+  width: 100%;
+`
+
+const DialogWrapper = styled(Box)`
+  display: table-cell;
+  vertical-align: middle;
+  ${props =>
+    props.enableoverflow &&
+    `
+    padding-top: 24px;
+    padding-bottom: 24px;
+  `};
+`
+
+const DialogInnerWrapper = styled(Flex)`
+  position: relative;
+  height: 100%;
+`
+
+const Modal = ({
+  isOpen,
+  onClose,
+  bg,
+  zIndex,
+  children,
+  imgMode,
+  width,
+  className,
+  header,
+  disableCloseButton,
+  enableOverflow,
+  height
+}) => (
+  <Transition
```

Comment:
@bertya I'm leaving a comment about this in Slack for you (contains a BB link, not apropos for GH)